eth_worker/eth_manager/blockchain_sync/blockchain_sync.py

```diff
@@ -0,0 +1,134 @@
+import json
+from uuid import uuid4
+import config
+import requests
+from requests.auth import HTTPBasicAuth
+from math import ceil
+from web3 import Web3
+
+import eth_manager.blockchain_sync.blockchain_sync_constants as sync_const
+from eth_manager import red, w3_websocket, persistence_module
+from eth_manager.ABIs import erc20_abi
+
+# Hit the database to get the latest block number to which we're synced
+def get_latest_block_number():
+    return w3_websocket.eth.getBlock('latest').number
+
+# Call app-level webhook with newfound transacitons
+def call_webhook(transaction):
+    body = {
+        'sender_blockchain_address': transaction.sender_address,
+        'recipient_blockchain_address': transaction.recipient_address,
+        'blockchain_transaction_hash': transaction.hash,
+        'transfer_amount': int(transaction.amount),
+        'contract_address': transaction.contract_address
+    }
+    r = requests.post(config.APP_HOST + '/api/v1/credit_transfer/internal/',
+                        json=body,
+                        auth=HTTPBasicAuth(config.INTERNAL_AUTH_USERNAME,
+                                        config.INTERNAL_AUTH_PASSWORD),
+                        timeout=30
+                    )
+    return r
+
+# Get list of filters from redis. This is the starting point of the synchronization process
+def synchronize_third_party_transactions():
```

Comment:
Called every 30s by the celery beat, so if execution is delayed, could have concurrency issues (more below).

Reply:
Added a redis lock, should now strictly run only one at a time